tools/generators/pbxproject_prefix/README.md

```diff
@@ -0,0 +1,52 @@
+# `PBXProject` prefix `PBXProj` partial generator
+
+The `pbxproject_prefix` generator creates a `PBXProj` partial containing the
+start (i.e. prefix) of the `PBXProject` element.
+
+## Inputs
+
+The generator accepts the following command-line arguments (see
+[`Arguments.swift`](src/Generator/Arguments.swift) and
+[`PBXProjectPrefix.swift`](src/PBXProjectPrefix.swift) for more details):
+
+- Positional `output-path`
+- Positional `workspace`
+- Positional `execution-root-file`
+- Positional `minimum-xcode-version`
+- Option `--development-region <development-region>`
+- Option `--organization-name <organization-name>`
+- Flag `--colorize`
+
+Here is an example invocation:
+
+```shell
+$ pbxproject_prefix \
+    /tmp/pbxproj_partials/pbxproject_prefix \
```

Comment:
I've never used `ArgumentParser`, is it common there to use positional args like this and not specify something like `--output-path` instead? Or that is an option and we're choosing to not make that the default usage?

Reply:
I'm only using flags like `--output-path` if required, to reduce Starlark memory usage. Times when they are required:

- Optional values
- To indicate the end of one list of values and the start of another

There is a slight readability loss, but the only place you should be constructing these are in the custom scheme definition (where there are comments), or in the Starlark code that creates the `args()` object (which will also have comments).

Reply:
>There is a slight readability loss, but the only place you should be constructing these are in the custom scheme definition (where there are comments), or in the Starlark code that creates the args() object (which will also have comments).

Ok, perfect, this addresses exactly what I was concerned about and agree that it should be fine then